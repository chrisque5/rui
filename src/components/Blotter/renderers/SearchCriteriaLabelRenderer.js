import { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import moment from 'moment';
import {
  Col, Layout, Row,
} from 'antd';
import { getBlotterSearchParams } from '../../../utils/selectors';

const SearchCriteriaLabelRenderer = () => {
  const searchParams = useSelector(getBlotterSearchParams);
  const [searchResultLabel, setSearchResultLabel] = useState('Search results for: ');
  const [dateBrokerSearchLabel, setDateBrokerSearchLabel] = useState(null);
  const [custTraderSearchLabel, setCustTraderSearchLabel] = useState(null);
  const [tooltipText, setTooltipText] = useState(null);
  const labelStyle = {
    maxWidth: 500, overflow: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap',
  };
  useEffect(() => {
    let dateBrokerSearch = 'TODAY';
    let custTraderSearch = '';
    const { dateFrom, dateTo, valid } = searchParams;
    let isToday = false;
    if (moment(dateFrom).isSame(moment(new Date()), 'day') && moment(dateFrom).isSame(moment(dateTo), 'day')) {
      isToday = true;
      setSearchResultLabel('Showing the results for: ');
    }
    if (valid) {
      const {
        dealId, broker, trader, customer,
      } = searchParams;

      if (!isToday || dealId || broker || trader || customer) {
        setSearchResultLabel('Search results for: ');
      }

      if (dealId) {
        dateBrokerSearch = `Deal ID: ${dealId}`;
        setCustTraderSearchLabel(custTraderSearch);
      } else {
        if (!isToday || broker || trader || customer) {
          dateBrokerSearch = `Trade Date From: ${moment(dateFrom).format('L')} To: ${moment(dateTo).format('L')}`;
        }

        if (broker) {
          dateBrokerSearch += `, Broker: ${broker} `;
        }

        if (customer) {
          custTraderSearch += `Customer: ${customer}`;
        }

        if (trader) {
          custTraderSearch += customer ? ', ' : '';
          custTraderSearch += `Trader: ${trader}`;
        }

        setCustTraderSearchLabel(custTraderSearch);
      }
    }
    setDateBrokerSearchLabel(dateBrokerSearch);
    let tooltip = `${dateBrokerSearch}`;
    if (custTraderSearch) {
      tooltip += `, ${custTraderSearch}`;
    }
    setTooltipText(tooltip);
  }, [searchParams]);

  return (
    <Layout title={tooltipText}>
      <Row>
        <Col>
          <span data-testid="lblSearchResult" className="ag-name-value">{searchResultLabel}</span>
        </Col>
        <Col style={{ ...labelStyle }}>
          <span data-testid="lblDateBrokerSearch" className="ag-name-value-value">
            {dateBrokerSearchLabel}
          </span>
        </Col>
      </Row>
      <Row>
        <Col style={{ width: 106 }} />
        <Col style={{ ...labelStyle }}>
          <span data-testid="lblCustTraderSearch" className="ag-name-value-value">{custTraderSearchLabel}</span>
        </Col>
      </Row>
    </Layout>
  );
};

export default SearchCriteriaLabelRenderer;
