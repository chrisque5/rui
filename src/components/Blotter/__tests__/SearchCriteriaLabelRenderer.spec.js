import {
  renderWithProvider, waitFor,
} from 'test-utils/testUtils';

import SearchCriteriaLabelRenderer from '../renderers/SearchCriteriaLabelRenderer';

describe('<SearchCriteriaLabelRenderer />', () => {
  let initialState = { };

  test('renders snapshot as expected when no last updated date set in state', () => {
    const { container } = renderWithProvider(
      <SearchCriteriaLabelRenderer />, { initialState },
    );
    expect(container).toMatchSnapshot();
  });

  test('verify showing label as TODAY if no search params', async () => {
    const { getByText } = renderWithProvider(<SearchCriteriaLabelRenderer />, { initialState });
    const todayText = await waitFor(() => getByText('TODAY'));
    expect(todayText).toBeTruthy();
    const searchLabel = await waitFor(() => getByText('Showing the results for:'));
    expect(searchLabel).toBeTruthy();
  });

  test('verify showing label for Deal ID search', async () => {
    initialState = {
      blotter: {
        searchParams: {
          valid: true,
          dealId: 1234,
        },
      },
    };
    const { getByText } = renderWithProvider(<SearchCriteriaLabelRenderer />, { initialState });
    const dealIdText = await waitFor(() => getByText('Deal ID: 1234'));
    expect(dealIdText).toBeTruthy();
    const searchLabel = await waitFor(() => getByText('Search results for:'));
    expect(searchLabel).toBeTruthy();
  });

  test('verify showing label for same date with broker search', async () => {
    const date = new Date('03 March 2021 15:49:11 UTC');
    initialState = {
      blotter: {
        searchParams: {
          valid: true,
          dateFrom: date,
          dateTo: date,
          broker: 'TEST BROKER',
        },
      },
    };

    const { getByText } = renderWithProvider(<SearchCriteriaLabelRenderer />, { initialState });
    const searchCriteriaText = await waitFor(() => getByText('Trade Date From: 03/03/2021 To: 03/03/2021, Broker: TEST BROKER'));
    expect(searchCriteriaText).toBeTruthy();
    const searchLabel = await waitFor(() => getByText('Search results for:'));
    expect(searchLabel).toBeTruthy();
  });

  test('verify showing label for different date with customer, trader search', async () => {
    initialState = {
      blotter: {
        searchParams: {
          valid: true,
          dateFrom: new Date('02 March 2021 15:49:11 UTC'),
          dateTo: new Date('03 March 2021 15:49:11 UTC'),
          trader: 'ABC',
          customer: 'DEF',
        },
      },
    };

    const { getByText } = renderWithProvider(<SearchCriteriaLabelRenderer />, { initialState });
    const searchCriteriaText = await waitFor(() => getByText('Customer: DEF, Trader: ABC'));
    expect(searchCriteriaText).toBeTruthy();
    const searchLabel = await waitFor(() => getByText('Search results for:'));
    expect(searchLabel).toBeTruthy();
  });
});
