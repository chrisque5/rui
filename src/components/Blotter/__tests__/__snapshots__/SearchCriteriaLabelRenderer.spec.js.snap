// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCriteriaLabelRenderer /> renders snapshot as expected when no last updated date set in state 1`] = `
<div>
  <section
    class="ant-layout"
    title="TODAY"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col"
      >
        <span
          class="ag-name-value"
          data-testid="lblSearchResult"
        >
          Showing the results for: 
        </span>
      </div>
      <div
        class="ant-col"
        style="max-width: 500px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
      >
        <span
          class="ag-name-value-value"
          data-testid="lblDateBrokerSearch"
        >
          TODAY
        </span>
      </div>
    </div>
    <div
      class="ant-row"
    >
      <div
        class="ant-col"
        style="width: 106px;"
      />
      <div
        class="ant-col"
        style="max-width: 500px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
      >
        <span
          class="ag-name-value-value"
          data-testid="lblCustTraderSearch"
        />
      </div>
    </div>
  </section>
</div>
`;
